enterprise/server/invocation_search_service/invocation_search_service.go

```diff
@@ -174,15 +174,36 @@ func (s *InvocationSearchService) QueryInvocations(ctx context.Context, req *inp
 	// Always add permissions check.
 	addPermissionsCheckToQuery(u, q)
 
-	sort := defaultSortParams()
-	if req.Sort != nil && req.Sort.SortField != inpb.InvocationSort_UNKNOWN_SORT_FIELD {
-		sort = req.Sort
+	sort := req.Sort
+	if sort == nil {
+		sort = defaultSortParams()
+	} else if req.Sort.SortField == inpb.InvocationSort_UNKNOWN_SORT_FIELD {
+		sort.SortField = defaultSortParams().SortField
 	}
 	switch sort.SortField {
 	case inpb.InvocationSort_CREATED_AT_USEC_SORT_FIELD:
 		q.SetOrderBy("created_at_usec", sort.Ascending)
 	case inpb.InvocationSort_UPDATED_AT_USEC_SORT_FIELD:
 		q.SetOrderBy("updated_at_usec", sort.Ascending)
+	case inpb.InvocationSort_DURATION_SORT_FIELD:
```

Comment:
I'm a bit concerned about the query performance, since these fields are not indexed. Have you profiled these queries or tested against our biggest customers?

Reply:
Thanks for pointing that out. In the common case (no sorting), the query to the database will be the same, so that shouldn't be affected by this PR. I suspect you're right that some of the other queries will be slower, but I don't think we should block this PR because this is a new feature and we can look into speeding it up if it proves to be a problem. Does that seem reasonable?